src/blocking/executor.rs

```diff
@@ -113,32 +83,32 @@ impl Executor {
         match self.retention_mode {
             RetentionMode::KeepAll => {
                 match result {
-                    Ok(task) => Queue::finish_task_query(&self.pooled_connection, &task).unwrap(),
-                    Err((task, error)) => {
-                        Queue::fail_task_query(&self.pooled_connection, &task, error).unwrap()
-                    }
+                    Ok(task) => self
+                        .queue
+                        .update_task_state(&task, FangTaskState::Finished)
+                        .unwrap(),
+                    Err((task, error)) => self.queue.fail_task(&task, error).unwrap(),
                 };
             }
             RetentionMode::RemoveAll => {
                 match result {
-                    Ok(task) => Queue::remove_task_query(&self.pooled_connection, task.id).unwrap(),
-                    Err((task, _error)) => {
-                        Queue::remove_task_query(&self.pooled_connection, task.id).unwrap()
-                    }
+                    Ok(task) => self.queue.remove_task(task.id).unwrap(),
+                    Err((task, _error)) => self.queue.remove_task(task.id).unwrap(),
                 };
             }
             RetentionMode::RemoveFinished => match result {
                 Ok(task) => {
-                    Queue::remove_task_query(&self.pooled_connection, task.id).unwrap();
+                    self.queue.remove_task(task.id).unwrap();
                 }
                 Err((task, error)) => {
-                    Queue::fail_task_query(&self.pooled_connection, &task, error).unwrap();
+                    self.queue.fail_task(&task, error).unwrap();
                 }
             },
         }
     }
 }
 
+//TODO fix tests
```

Comment:
let's not add such comments. you can create a list of todos in the PR's description

Reply:
deleted in [Finishing blocking module, starting to modify tests](https://github.com/ayrat555/fang/pull/74/commits/7d1402f52f94e1d0520746e15693116b5988ed4f)